### ISSUE TITLE
Memory leak on iOS

### ISSUE THREAD

**Issue by arafeek** [reneweb/react-native-tensorflow#14]
Seems that there is a memory leak in the iOS version of the library.  

On subsequent uses of the library we can see memory usage steadily increase until the app eventually crashes.

My setup is something like below:

```
function categorizeImage(imagePath) {
  const processor = new TfImageRecognition({
    model: require('my-model.pb'),
    labels: require('my-labels.txt'),
  });
  return processor.recognize({ image: imagePath, ..... });
}
```

Each time `categorizeImage` is called we see memory usage increase. Images provided below.

![snip20180206_27](https://user-images.githubusercontent.com/7095609/35882707-0f22117a-0b53-11e8-8adc-23faf6a5fe7e.png)
![snip20180206_28](https://user-images.githubusercontent.com/7095609/35882716-14153964-0b53-11e8-96ea-0583e02f643d.png)

**Comment by tfsjohan** [comment]
Does it help to create the processor outside of the function and store as class member instead? Should be faster too since you only need to init it once.

**Comment by arafeek** [comment]
Tried that, but then on the second call to `recognize()` It throws an error saying I fed the input more than once.

**Comment by reneweb** [comment]
Hey, I've looked into it and there where a few issues related to memory usage. Primarily there is a close function on the native side, which wasn't properly exposed (and documented) on the js side. I've just pushed a fix for this. If you update to version 0.1.3 you should have a close function after you called the `recognize` function. I've also updated the docs accordingly.
Also good to now that there is an issue with reusing the `recognize` function using the same `TfImageRecognition`. I will raise an issue to look into that.

**Comment by reneweb** [comment]
I've now pushed a fix that allows to call `recognize` multiple times using the same tf instance and it is advisable to avoid re-instantiating the tf obj unnecessarly as it will load the model and label each time. The fix is in version 0.1.5 (issue is #17 )

**Comment by arafeek** [comment]
Awesome, thanks!
